### ISSUE TITLE
Guidelines: WCAG 2.0 (Level AA) not followed for ngb date-picker

### ISSUE THREAD

**Issue by kedar9444** [ng-bootstrap/ng-bootstrap#2543]
### Bug description: The auto generated select fields for month and year violates the **WCAG 2.0 (Level AA) guidelines**.

Error code : Check 91: select element missing an associated label.
Repair: Add a label element that surrounds the control's label. Set the for attribute on the label element to the same value as the id attribute of the control. And/or add a title attribute to the input element. And/or create a label element that contains the input element.

since the html is auto-generated we can not add the label for it.

### Version of Angular, ng-bootstrap, and Bootstrap:

**Angular: "^5.1.3"**

**ng-bootstrap:  "1.1.0"**

**Bootstrap: "4.0.0"**

**Comment by kedar9444** [comment]
website used for checking : https://achecker.ca/

you can use the html from this link to reproduce the error : 
https://ng-bootstrap.github.io/#/components/datepicker/overview

**Comment by maxokorokov** [comment]
An option would be to add a title something like:

```html
<select title="months"></select>
<select title="years"></select>
```

But these strings should be added to i18n in this case

**Comment by kedar9444** [comment]
@maxokorokov :  **But these strings should be added to i18n in this case**?
 i am not clear about this step can you please elaborate it further.

**Comment by kedar9444** [comment]
This is what my code looks like:

![image](https://user-images.githubusercontent.com/20212215/43083498-4ae6f100-8eb4-11e8-8a53-74500b0d9810.png)

Add this is how it appears : 
![image](https://user-images.githubusercontent.com/20212215/43083531-627a6eb4-8eb4-11e8-9b05-7a2601f47ce9.png)

and here is dynamically generated DOM elements for date-picker, which are clearly missing labels for select boxes used for month and year selection : 

![image](https://user-images.githubusercontent.com/20212215/43083709-ca726bb6-8eb4-11e8-910c-aed62d2a6119.png)

which violates the WCAG 2.0 guild lines  which states : **Check 91: select element missing an associated label**

**Comment by maxokorokov** [comment]
@kedar9444, sure, I see the problem, it was clear from the description. Adding the `title` attribute should fix it.

> But these strings should be added to i18n in this case

Just wanted to say that if we put a hardcoded string as a select box title, we should not forget to translate it.

**Comment by kedar9444** [comment]
@maxokorokov : yes, got it.
do you guys have any plan to add/implement these WCAG guidelines for ng-bootstrap?

**Comment by pkozlowski-opensource** [comment]
@maxokorokov `title="..."` would add a "tooltip" to a `<select>` element, right?
Maybe it is enough to add `aria-label="..."` if we don't want to have this "tooltip" visually / functionally speaking?

**Comment by aakash1401** [comment]
@pkozlowski-opensource  how we plan to add area-label or title to dynamically generated 'SELECT' element. Because we can't do it from HTML as we can only add <ngb-datepicker> and available inputs to pass. 
Are you planning it for upcoming releases or next versions?

**Comment by maxokorokov** [comment]
> website used for checking : https://achecker.ca/

By the way, adding `aria-label` still reported an error for me with the website in question

**Comment by kedar9444** [comment]
@maxokorokov : adding title along with aria-label solves the problem for achecker.ca

**Comment by pkozlowski-opensource** [comment]
Fixed through #2636, will be part of a next release.
